### PR TITLE
[Distributed] Refactor distributed training to new DGLGraph.

### PR DESCRIPTION
**Pull request by zheng-da**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

File: tests/distributed/test_partition.py [hunk 1]
```diff
@@ -14,13 +14,16 @@
 
 def create_random_graph(n):
     arr = (spsp.random(n, n, density=0.001, format='coo', random_state=100) != 0).astype(np.int64)
-    ig = create_graph_index(arr, readonly=True)
+    arr = arr.tocoo()
+    row = F.tensor(arr.row, dtype=F.int64)
+    col = F.tensor(arr.col, dtype=F.int64)
+    ig = create_unitgraph_from_coo(1, n, n, row, col, ['coo', 'csr', 'csc'])
     return dgl.DGLGraph(ig)
```
Comment (VoVAllen) [inline tests/distributed/test_partition.py:21]: directly use dgl.graph(arr) should work

File: python/dgl/distributed/graph_services.py [hunk 2]
```diff
@@ -133,7 +133,7 @@ def merge_graphs(res_list, num_nodes):
         dst_tensor = res_list[0].global_dst
         eid_tensor = res_list[0].global_eids
     g = graph((src_tensor, dst_tensor),
-              restrict_format='coo', num_nodes=num_nodes)
+              formats=('coo',), num_nodes=num_nodes)
```
Comment (zheng-da) [inline python/dgl/distributed/graph_services.py:136]: do we need to specify formats here? @jermainewang the graph here is a subgraph generated from sampling.

Comment (yzh119) [reply python/dgl/distributed/graph_services.py]: You can compare the speed of restricting `coo` and not restricting format after minjie finished the refactor the scheduler.

Comment (yzh119) [reply python/dgl/distributed/graph_services.py]: btw `formats` also accepts a single string as input: `formats='coo'`

Comment (zheng-da) [reply python/dgl/distributed/graph_services.py]: does it work?

Comment (zheng-da) [reply python/dgl/distributed/graph_services.py]: i see. it doesn't work in `shared_memory`.

File: python/dgl/distributed/dist_graph.py [hunk 3]
```diff
@@ -85,11 +80,10 @@ def _get_graph_from_shared_mem(graph_name):
...
     '''
-    gidx = from_shared_mem_graph_index(_get_graph_path(graph_name))
-    if gidx is None:
-        return gidx
-
-    g = DGLGraph(gidx)
+    g, ntypes, etypes = heterograph_index.create_heterograph_from_shared_memory(graph_name)
+    if g is None:
+        return None
+    g = DGLHeteroGraph(g, ntypes, etypes)
```
Comment (VoVAllen) [inline python/dgl/distributed/dist_graph.py:86]: I think you can do this by `dgl.hetero_from_shared_memory` at https://github.com/soodoshll/dgl/blob/a7dbfd1341d30125e9f740f514aad10e34325510/python/dgl/convert.py#L418

File: python/dgl/distributed/dist_graph.py [hunk 4]
```diff
@@ -315,11 +309,7 @@ def __init__(self, ip_config, graph_name, gpb=None, conf_file=None):
         else:
             connect_to_server(ip_config=ip_config)
             self._client = KVClient(ip_config)
-            g = _get_graph_from_shared_mem(graph_name)
-            if g is not None:
-                self._g = as_heterograph(g)
-            else:
-                self._g = None
+            self._g = _get_graph_from_shared_mem(graph_name)
```
Comment (VoVAllen) [inline python/dgl/distributed/dist_graph.py:312]: Why is this still `_get_graph_from_shared_mem`?

File: python/dgl/distributed/graph_services.py (hunk 2 above)
Comment (VoVAllen) [reply python/dgl/distributed/graph_services.py]: Why it doesn't work with shared_memory?

File: python/dgl/distributed/dist_graph.py (hunk 3 above)
Comment (zheng-da) [reply python/dgl/distributed/dist_graph.py:86]: it's different. i need to deal with the case that shared-memory graph index doesn't exist. also i don't want to import from convert.py.

File: python/dgl/distributed/dist_graph.py (hunk 4 above)
Comment (zheng-da) [reply python/dgl/distributed/dist_graph.py:312]: i don't just share graph index. i also share some node/edge data. please check the function.

**Pull request merged**